controlpanel/api/rules.py

```diff
@@ -0,0 +1,194 @@
+"""
+Rules and permissions (and object permissions) using django-rules
+"""
+
+from rules import add_perm, is_active, is_authenticated, is_superuser, predicate
```

Comment:
Probably a stupid question, where are these funtions defined?

Reply:
These function are imported from `rules`, which is the [Django Rules library](https://github.com/dfunckt/django-rules) (eg: [is_superuser](https://github.com/dfunckt/django-rules/blob/46c594ec2abb605647af49bfbbca17326c3f9df3/rules/predicates.py#L275))